### ISSUE TITLE
🐛 BUG: wrangler should fail if secrets put is used in a pages project

### ISSUE THREAD

**Issue by divyenduz** [cloudflare/workers-sdk#6217]
### Which Cloudflare product(s) does this pertain to?

Pages

### What version(s) of the tool(s) are you using?

3.57.1

### What version of Node are you using?

v18.18.1

### What operating system and version are you using?

Mac Sonoma 14.5 (23F79)

### Observed behavior
`wrangler secret put API_KEY` in a pages project creates a worker and puts the secret in the worker.

### Expected behavior
Either the `wrangler secret put API_KEY` command should create the secret in the relevant pages project or fail. Creating a new worker is incorrect. 

### Steps to reproduce
Please provide the following:
- `wrangler.toml`

```toml
name = "<project name>"
compatibility_date = "2024-07-01"
pages_build_output_dir = "./build/client"

compatibility_flags = [ "nodejs_compat" ]
```

- run `npx wrangler secret put API_KEY`, this command succeeds but it should fail in a pages project. This command creates a brand new works and puts the secret there. This is not intuitive. 

Here is the internal discord chat with the related context: https://discord.com/channels/595317990191398933/799437470004412476/1258446892564484218

**Comment by CarmenPopoviciu** [comment]
hi @divyenduz,

thank you for raising this issue with us.

The command you are looking for is `wrangler pages secret put`, see [here](https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/pages/secret/index.ts#L113-L114). `wrangler secret put` work for Workers only! 

Can you please give that a try and let me know if you're still experiencing issues? 🙏 

I also just noticed we don't have this command documented :/. Going to fix that in the meanwhile!

**Comment by divyenduz** [comment]
Thank you, still the original issue remains that ‘wrangler secret put API_KEY’ should fail in a pages project or redirect to the correct pages command. 

I will give this command a go! I unblocked myself by setting the secrets from the UI

**Comment by CarmenPopoviciu** [comment]
I've updated our official docs with the missing information abt the `wrangler pages secret` command. These changes are already live at https://developers.cloudflare.com/workers/wrangler/commands/#pages

Another way for folks to see all commands available and supported by Pages, is via `wrangler pages help`. Running this command will output the following:

![Image](https://github.com/user-attachments/assets/59b3e9b1-e8d6-4a9c-bdbc-7cfcfd1d097c)

I think in this particular case it was very unfortunate that `pages secret` was not documented in our official docs 💔 . I believe this is what caused the confusion here, and what possibly led folks to run `wrangler secret` in a Pages project? That it didn't throw an error, is also very unfortunate, but merely a coincidence, as proven by, the rightfully so, unexpected behaviour.

The fact remains that `wrangler secret` in Pages is kind of a "wrong tool in the wrong place" situation. It is not a Pages command, and it is not expected to work, or even be run in the context of a Pages project, just like for example `wrangler deploy` isn't either. I can absolutely appreciate if these things are confusing, and it is precisely this sort of use cases that the Workers <> Pages convergence will solve! In the meanwhile, if we are not being clear enough about these differences in our docs, we should definitely do better, and I'd like to hear more about that, and where this specific confusion started from, so we can fix the issues. But IMHO, the fix should not be trying to detect which command is being executed in which type of project, since that will open a pandora's box of checks and corner cases.

This issue was closed automatically due to the docs PR merge, but feel free to let me know if `wrangler pages secret` did the trick, and if the docs changes help clarify things a bit further.

Thanks again for raising this issue. While not intended as such, it was a great callout to our incomplete docs 🙏
